state_transfer/transfer/rsync/server.go

```diff
@@ -91,6 +94,14 @@ func createRsyncServerResources(c client.Client, r *RsyncTransfer, ns string) er
 
 func createRsyncServerConfig(c client.Client, r *RsyncTransfer, ns string) error {
 	var rsyncConf bytes.Buffer
+	runRsyncRoot := false
+	for _, ops := range r.options.DestContainerMutations {
```

Comment:
What about source container mutations?

Reply:
Source container mutations don't matter, right?. cause eventually in destination cluster file permissions would be the same as "writing" process/pod. And since our server Rsync pods will run in destination (4.11/4.12/onwards) we would need to change server-side only.

Reply:
@JaydipGabani what happens when source is a 4.12+ cluster?

Reply:
If we support 4.12+, we support it as source and destination both

Reply:
@pranavgaikwad I think that change is needed on mig-controller side, we are not creating any client-side config, so to make the client work in 4.12+ environment we would need to update pod capabilities and flags for rsync.